### PR TITLE
[ZEPPELIN-5455] Spark conf can not contain white space

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR would use `|` as parameter delimiter, although it can not resolve the issue fundamentally, because parameter may still contain `|`, but it would mitigate the issue a lot. 

### What type of PR is it?
[Improvement ]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5455

### How should this be tested?
* Manually tested

### Screenshots (if appropriate)
image

### Questions:
* Does the licenses files need update? NO
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by Reamer** [comment]
@zjffdu 
Please take a look at #4127 and my [comment](https://github.com/apache/zeppelin/pull/4127#issuecomment-864825992). I tried to remove IFS, which is not correct for word splitting.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public SparkInterpreterLauncher(ZeppelinConfiguration zConf, RecoveryStorage rec
               " to false if you want to use other modes.");
     }
 
-    StringBuilder sparkConfBuilder = new StringBuilder();
+    List<String> sparkConfList = new ArrayList<>();
```
Comment (Reamer) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java:109]: A StringJoiner is more elegant

https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html

Comment (zjffdu) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java]: Thanks for the review, PR is updated, `SparkInterpreterLauncher` & `FlinkInterpreterLauncher` both use StringJoiner now.

**Comment by zjffdu** [comment]
Will merge if no more comment

**Pull request closed without merging**